### PR TITLE
Jabra-gnp: Adds support to OTA updates. Includes some bug fixes

### PR DESCRIPTION
**Pull request by gdpcastro**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by gdpcastro** [comment]
I could maybe use some advice regarding the "framework" around child devices - as I went with what seemed the most reasonable, but I do get a small issue currently. After the update is done on a child device (that is a device connected to one of our dongles), I get a message from fwupdtool "device version not updated on success, V1 != V2". 
The version is updated, but I imagine fwupdtool queries for the new version before the device reconnects itself.

**Comment by hughsie** [comment]
@gdpcastro I think the `device version not updated on success` is because the parent and the child have the same device-id -- I think you meant to set the *logical* ID to ota_device.

I've pushed a fixup and probably broken a few things, but be helpful in showing you the way to do this. I hope it's helpful!

**Comment by gdpcastro** [comment]
I'm going to be on vacation for some days, so this will be on hold for around a week

**Comment by gdpcastro** [comment]
> @gdpcastro I think the `device version not updated on success` is because the parent and the child have the same device-id -- I think you meant to set the _logical_ ID to ota_device.

I still get `device version not updated on success, 1.7.4 != 1.8.13` - might be a problem with the fact that the ota device takes a second to re-pair to the dongle. Not entirely sure how to address it properly. I did try adding some waiting times, but to no avail

**Comment by hughsie** [comment]
> might be a problem with the fact that the ota device takes a second to re-pair to the dongle

I think we're missing a `fu_device_add_flag(child, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);` somewhere -- i.e. just before the child device "goes away" and "comes back" from the daemon. If the child device never really goes away, then `fu_device_sleep()` in `->reload()` might be your friend.

File: plugins/jabra-gnp/fu-jabra-gnp-device.c [hunk 1]
```diff
@@ -104,24 +118,34 @@ fu_jabra_gnp_device_rx_cb(FuDevice *device, gpointer user_data, GError **error)
...
+						     NULL, /* cancellable */
+						     error)) {
+			g_prefix_error(error, "failed to read from device: ");
+			return FALSE;
 		}
 	}
 
+	if (fu_memcmp_safe((const guint8 *)rx_data->rxbuf,
+			   FU_JABRA_GNP_BUF_SIZE,
+			   0,
```
Comment (hughsie) [inline plugins/jabra-gnp/fu-jabra-gnp-device.c:140]: this doesn't fill me with the warm-and-fuzzies; could we do something like:

     typedef struct {
    -       guint8 *rxbuf;
    +       guint8 rxbuf[FU_JABRA_GNP_BUF_SIZE];
            const guint timeout;
     } FuJabraGnpRxData;

...and then do `sizeof(rx_data->rxbuf)` here? then we can also get rid of a few local buffers too. Same comment for FuJabraGnpTxData obviously too.

File: plugins/jabra-gnp/fu-jabra-gnp-device.c [hunk 2]
```diff
@@ -608,16 +731,87 @@ fu_jabra_gnp_device_probe(FuDevice *device, GError **error)
...
 	if (!fu_jabra_gnp_device_read_version(self, error))
 		return FALSE;
 	if (!fu_jabra_gnp_device_read_dfu_pid(self, error))
 		return FALSE;
+	if (self->address == FU_JABRA_GNP_ADDRESS_PARENT) {
+		guint16 child_dfu_pid = 0;
+		if (!fu_jabra_gnp_device_read_child_dfu_pid(self, &child_dfu_pid, &error_local)) {
+			g_debug("unable to read child's PID, %s", error_local->message);
+			return TRUE;
+		} else {
```
Comment (hughsie) [inline plugins/jabra-gnp/fu-jabra-gnp-device.c:803]: I don;t think you need the }else{ here

**Review by hughsie — APPROVED** [review]
@gdpcastro I've made a few cosmetic changes, and squashed everything down. Can you test one more time and then please merge. Thanks!

**Comment by gdpcastro** [comment]
Tests went through :) Merging!

**Comment by hughsie** [comment]
@sorenkrabbe this was merged to main which is what's going to become `2.0.0` -- if you need it in 1.9.x then that's the `1_9_X` branch. 2.0.0 is due for the end of the month -- is that any good for you?

**Comment by sorenkrabbe** [comment]
@hughsie. Thanks. That explains it.
Waiting until end of month should be fine. However, what is the plan the for 1.9.x version? Do you expect there will be upstream projects you'll continue to serve with 1.9.x releases after 2.x has released or will their upgrade path be to go 2.x?

**Comment by hughsie** [comment]
> However, what is the plan the for 1.9.x version

It's being shipped in RHEL for the next 10 years! :) Do you need it backporting? Is that something you want to do, or do you need a hand?

**Comment by sorenkrabbe** [comment]
Thanks. A hand with the backporting would be highly appreciated.

**Pull request merged**